File: typescript/angular2/components/button/src/utilities.service.ts
import {Injectable} from 'angular2/core';

"use strict";

@Injectable()
export class Utilities {
	public static forEach(
		object: Object,
		callback: (value: any, key: string, object: Object) => void,
		properties: Array<string> = null): void {
		// Execute callback for provided properties of object
		for(var key in object) {
			if(	object.hasOwnProperty(key)
				&& (properties === null || properties.indexOf(key) !== -1)) {
				callback(object[key], key, object);
			}
		}
	}
}

File: dart/angular2/lib/components/button/utilities_service.dart
library angular2_dart_examples.utilities_service;

typedef dynamic ForEachCallback(dynamic value, [String key, Map<String, dynamic> object]);

class Utilities {
	static void forEach(Map<String, dynamic> object, ForEachCallback callback, [List<String> properties]) {
		void filter(String key, dynamic value) {
			if (properties == null || properties.contains(key)) {
				callback(value, key, object);
			}
		}
		// Execute callback for provided properties of object
		object.forEach(filter);
	}
}